### PR TITLE
Check for stdlibs in REQUIRE

### PR DESCRIPTION
**Pull request by ararslan**
This adds a check that will error if a package lists a stdlib module in its REQUIRE file.

### CODE REVIEW

File: .test/check-pr.jl [hunk 1]
```diff
@@ -189,3 +189,46 @@ if VERSION >= v"0.7.0"
...
+    "Logging",
+    "Markdown",
+    "Mmap",
+    "Pkg",
+    "Printf",
+    "Profile",
+    "Random",
+    "REPL",
+    "Serialization",
+    "SHA",
```
Comment (oxinabox) [inline .test/check-pr.jl]: Are we still allowing 0.6 packages to update their Metadata? If so banning them from depending on SHA.jl is not going to be ok.

File: .test/check-pr.jl [hunk 2]
```diff
@@ -189,3 +189,46 @@ if VERSION >= v"0.7.0"
...
+    "SharedArrays",
+    "Sockets",
+    "SparseArrays",
+    "Statistics",
+    "SuiteSparse",
+    "Test",
+    "Unicode",
+    "UUIDs",
+]
```
Comment (omus) [inline .test/check-pr.jl]: Probably better to use Pkg3 to get this list? That would mean this check couldn't occur on Julia 0.6 checks

**Comment by ararslan** [comment]
Concerns addressed. The list of stdlib modules is now obtained directly from Pkg, and this allows SHA in REQUIRE files so long as the minimum required Julia version is before it was added as a stdlib.

**Comment by andreasnoack** [comment]
Great. Thanks for adding this check.

**Pull request merged**